app/controllers/api/base_controller.rb

```diff
@@ -225,6 +226,24 @@ def paginate_options
       }
     end
 
+    def assign_lone_taxonomies
+      return unless resource_class_for(resource_name)
+      [Location, Organization].each do |taxonomy|
+        tax_name = taxonomy.to_s.downcase
+        if resource_class.reflections.has_key? tax_name.pluralize
+          tax_ids = "#{tax_name}_ids"
+          next if params[resource_name].try(:has_key?, tax_ids)
+          next unless taxonomy.one?
+          params[resource_name][tax_ids] = [taxonomy.first.id]
+        elsif resource_class.reflections.has_key? tax_name
+          tax_id = "#{tax_name}_id"
+          next if params[resource_name].try(:has_key?, tax_id)
+          next unless taxonomy.one?
```

Comment:
This check might trigger a DB call, so I prefer to repeat the code and only run it after previous cheaper checks that may short-circuit in many cases rather than run it twice on all api requests.